dev/run-tests.py

```diff
@@ -0,0 +1,417 @@
+
+        # find any sql files
+        sql_files = [f for f in changed_files
+                     if multi_starts_with(f, 
+                                          "sql/", 
+                                          "bin/spark-sql", 
+                                          "sbin/start-thriftserver.sh")]
+
+        non_sql_files = set(changed_files).difference(set(sql_files))
+
+        if non_sql_files:
+            test_suite.append("CORE")
+        if sql_files:
+            print "[info] Detected changes in SQL. Will run Hive test suite."
+            test_suite.append("SQL")
+            if not non_sql_files:
+                print "[info] Detected no changes except in SQL. Will only run SQL tests."
+        return set(test_suite)
+    else:
+        # we aren't in the Amplab environment so merely run all tests
+        test_suite.append("CORE")
+        test_suite.append("SQL")
+        return set(test_suite)
+
+
+def set_title_and_block(title, err_block):
+    os.environ["CURRENT_BLOCK"] = error_codes[err_block]
+    line_str = '=' * 72
+
+    print
+    print line_str
+    print title
+    print line_str
+
+
+def run_apache_rat_checks():
+    set_title_and_block("Running Apache RAT checks", "BLOCK_RAT")
+    run_cmd(["./dev/check-license"])
+
+
+def run_scala_style_checks():
+    set_title_and_block("Running Scala style checks", "BLOCK_SCALA_STYLE")
+    run_cmd(["./dev/lint-scala"])
+
+
+def run_python_style_checks():
+    set_title_and_block("Running Python style checks", "BLOCK_PYTHON_STYLE")
+    run_cmd(["./dev/lint-python"])
+
+
+def exec_maven(mvn_args=[]):
+    """Will call Maven in the current directory with the list of mvn_args passed
+    in and returns the subprocess for any further processing"""
+
+    run_cmd(["./build/mvn"] + mvn_args)
+
+
+def exec_sbt(sbt_args=[]):
+    """Will call SBT in the current directory with the list of mvn_args passed
+    in and returns the subprocess for any further processing"""
+
+    sbt_cmd = ["./build/sbt"] + sbt_args
+
+    # NOTE: echo "q" is needed because sbt on encountering a build file
+    # with failure (either resolution or compilation) prompts the user for
+    # input either q, r, etc to quit or retry. This echo is there to make it
+    # not block.
+    echo_proc = subprocess.Popen(["echo", "\"q\n\""], stdout=subprocess.PIPE)
+    sbt_proc = subprocess.Popen(sbt_cmd,
+                                stdin=echo_proc.stdout,
+                                stdout=subprocess.PIPE)
+    echo_proc.wait()
+    for line in iter(sbt_proc.stdout.readline, ''):
+        if not SBT_OUTPUT_FILTER.match(line):
+            print line,    
+    retcode = sbt_proc.wait()
+
+    if retcode > 0:
+        exit_from_command_with_retcode(sbt_cmd, retcode)
+
+
+def build_apache_spark():
+    """Will first build Spark with Hive v0.12.0 to ensure the build is
+    successful and, after, will build Spark again against Hive v0.13.1 as the
+    tests are based off that"""
```

Comment:
We are no longer doing two different builds for hive 12 and 13, so I think this comment is outdated.